homeassistant/components/knx/schema.py

```diff
@@ -724,17 +742,24 @@ class SensorSchema(KNXPlatformSchema):
 
     CONF_ALWAYS_CALLBACK = "always_callback"
     CONF_STATE_ADDRESS = CONF_STATE_ADDRESS
+    CONF_STATE_CLASS = "state_class"
     CONF_SYNC_STATE = CONF_SYNC_STATE
     DEFAULT_NAME = "KNX Sensor"
 
-    ENTITY_SCHEMA = vol.Schema(
-        {
-            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-            vol.Optional(CONF_SYNC_STATE, default=True): sync_state_validator,
-            vol.Optional(CONF_ALWAYS_CALLBACK, default=False): cv.boolean,
-            vol.Required(CONF_TYPE): sensor_type_validator,
-            vol.Required(CONF_STATE_ADDRESS): ga_list_validator,
-        }
+    ENTITY_SCHEMA = vol.All(
+        vol.Schema(
+            {
+                vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+                vol.Optional(CONF_SYNC_STATE, default=True): sync_state_validator,
+                vol.Optional(CONF_ALWAYS_CALLBACK, default=False): cv.boolean,
+                vol.Optional(
+                    CONF_STATE_CLASS, default=STATE_CLASS_MEASUREMENT
```

Comment:
I don't think this should be default, as its not exclusively for numeric values.

Reply:
The only valid non-numeric type is "string" - and for this type "measurement" will be replaced with `None` in the sub_validator. 

I could add a check `isinstance(dpt_class, DPTNumeric)` to make super-sure, but I believe these values should be applicable for long time statistics without users having to change their config (adding the state class everywhere). 

`last_reset` for energy will be a different story.

Reply:
Both go hand in hand. If a metered sensor gets a measurement device class without a last_reset, the statistics will also be wrongly recorded.

So with a numeric check, you won't cover the difference between a current measurement or a metered measurement.

Reply:
My plan is to add a dedicated config option for metered values next that have support for bus-resets too. So `sensor:` is more for temperatures and power, not energy, water usage etc.
```yaml
knx:
  sensor:
    - name: "Outside Temperature"
      type: temperature
      state_address: 1/1/1
  meter:
    # this will create a SensorEntity with a service to reset the meter when the address is configured
    - name: "Energy Fridge"
      type: active_energy_kwh
      state_address: 2/2/2
      reset_address: 3/3/3  # to trigger resets and also listen to resets from the bus
      last_reset_state_address: 4/4/4  # to read last_reset timestamp if available - if not it will be set when reset_address receives a `True` telegram
```

Reply:
Sure, besides I think that is additional confusing (as Home Assistant itself doesn't have a `meter` type), currently those are sensors. Marking them default as "current measurements" is therefore wrong. As existing configurations will have, for example, a kwh sensor defined (that with this change now is classified as a current measurement).

Reply:
So you think we should add `reset_address:` and `last_reset_state_address:` directly to sensor? That would be fine for me too.
Not sure how to distinguish between types that aren't applicable for `last_reset` (eg. temperature) and sensors that would have `last_reset: timestamp(0)` because they don't get resetted (or even set `last_reset` when the value changes to 0). Maybe it requires an additional config option.

Reply:
> So you think we should add reset_address: and last_reset_state_address: directly to sensor? That would be fine for me too.

I dunno, that was not the scope of my review. My concern is the default, of marking it as a current measurement, while for existing configurations, it can be a metered value already. Thus applying this default, will cause incorrect statistics being recorded for existing setups.

Reply:
I've removed the default. But I still think having to add boilerplate to 99% of all KNX sensors is a bad tradeoff.

Reply:
Can't we just drop asking it from the user and just tie it to the KNX "type"? In your example `temperature` and `active_energy_kwh` would both be measurement?

Reply:
If we have a last reset we can definitely assume it's a measurement because you can't reset historical data.